enhancements/cluster-scope-secret-volumes/csi-driver-host-injections.md

```diff
@@ -0,0 +1,306 @@
+---
+title: inject-cross-namespace-secrets-onto-openshift-hosts-for-general-pod-consumption
+
+authors:
+  - "@gabemontero"
+
+reviewers:
+  - "@bparees"
+  - "@adambkaplan"
+  - "@derekwaynecarr"
+  - "@gnufied"
+  
+approvers:
+  - "@bparees"
+  - "@adambkaplan"
+  - "@derekwaynecarr"
+  - "@gnufied"
+  
+creation-date: 2020-03-17
+
+last-updated: 2020-03-18
+
+<!-- status: provisional|implementable|implemented|deferred|rejected|withdrawn|replaced -->
+status: provisional
+
+
+  
+---
+
+# Inject cross namespace secrets onto OpenShift Hosts for general Pod consumption
+
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [/] Design details are appropriately documented from clear requirements
+- [/] Test plan is defined
+- [/] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Open Questions [optional]
+
+> 1. Should we manipulate our CSI plugin via the API from the CSI Spec outside the context of K8S Pod provisioning.  For 
+>example, as part of a Operator/Controller co-located with the CSI plugin, can/should we call `DestroyVolme`, `NodeUnstageVolume`, or
+>`NodeUnpublishVolume` when that Operator/Controller detects a change in any watched Secret resources, as part of updating
+>the contents on the host/node?  Or should that Operator/Controller, as part of being hosted as a privileged `DaemonSet`, directly 
+>manipulate the predefined location for the host/node content mapped to the CSI driver?  Sense is the latter is fine,
+>but confirmation from CSI SMEs would be good.
+
+## Summary
+
+Customer feedback on V4.x of OpenShift has included the desire to seamlessly inject certain content (usually of a sensitive nature)
+onto each host/node of a cluster for inclusion in the customer's pod.
+
+At this time, the only means of doing so are 
+
+- Declare `HostPath` based PV/PVCs on the Pod spec, on a per Pod basis, by each author of the Pod
+- Leverage the 4.x machine configuration operator or MCO to update the host file system
+
+There are several drawbacks with these approaches:
+
+- `HostPath` volumes require the privileged bit being set on the Pod, which grants the Pod and Pod author overall permissions
+that cluster administrators will not want, instead of access to just the specific piece of content on the host file system.
+- Host injection via the MCO requires a node reboot for the updates to take effect.  That is generally deemed 
+undesirable for this feature.  
+- The MCO API has not been intuitive for the type of users to date who are interested in introducing and then consuming
+small collections of files on a host.
```

Comment:
Most of this section above should be moved to the Alternatives section.

Reply:
part of next push